src/spider/storage/DataStorage.hpp

```diff
@@ -32,6 +32,36 @@ class DataStorage {
             = 0;
     virtual auto get_data(StorageConnection& conn, boost::uuids::uuid id, Data* data) -> StorageErr
             = 0;
+    /**
+     * Get data and register data reference from a driver in a single transaction.
```

Comment:
```suggestion
     * Get a data object and register a reference for it from the given driver in a single
     * transaction.
```
Feel free to improve on this further.